test/unit/host-config.spec.ts

```diff
@@ -0,0 +1,701 @@
+                    redirector = foo;
+                }
+            `;
+			const conf = RedsocksConf.parse(redsocksConfStr);
+			expect(conf).to.deep.equal({});
+		});
+
+		it('parses `redsocks {...}` config block', () => {
+			const redsocksConfStr = stripIndent`
+                redsocks {
+                    local_ip = 127.0.0.1;
+                    local_port = 12345;
+                    type = http-connect;
+                    ip = test2.balenadev.io;
+                    port = 1082;
+                    login = "user";
+                    password = "pass";
+                }
+            `;
+			const conf = RedsocksConf.parse(redsocksConfStr);
+			expect(conf).to.deep.equal({
+				redsocks: {
+					local_ip: '127.0.0.1',
+					local_port: 12345,
+					type: 'http-connect',
+					ip: 'test2.balenadev.io',
+					port: 1082,
+					login: '"user"',
+					password: '"pass"',
+				},
+			});
+		});
+
+		// Allow invalid fields through for backwards compatibility
+		it('parses to redsocks config with warnings while any values are invalid', () => {
+			const redsocksConfStr = stripIndent`
+                redsocks {
+                    local_ip = 123;
+                    local_port = foo;
+                    type = socks6;
+                    ip = 456;
+                    port = bar;
+                    login = user;
+                    password = pass;
+                }
+            `;
+			(log.warn as SinonStub).resetHistory();
+			const conf = RedsocksConf.parse(redsocksConfStr);
+			expect((log.warn as SinonStub).lastCall.args[0]).to.equal(
+				'Unexpected redsocks.conf configuration detected, proxy may not behave as expected:\n' +
+					'Expecting NumericIdentifier at 0.0.port but instead got: "bar" (must be be an positive integer)\n' +
+					'Expecting one of:\n' +
+					'    "socks4"\n' +
+					'    "socks5"\n' +
+					'    "http-connect"\n' +
+					'    "http-relay"\n' +
+					'at 0.0.type but instead got: "socks6"\n' +
+					'Expecting DoubleQuotedString at 0.1.login but instead got: <REDACTED> (must be enclosed in double quotes)\n' +
+					'Expecting DoubleQuotedString at 0.1.password but instead got: <REDACTED> (must be enclosed in double quotes)\n' +
+					'Expecting NumericIdentifier at 1.local_port but instead got: "foo" (must be be an positive integer)',
+			);
+			(log.warn as SinonStub).resetHistory();
+			expect(conf).to.deep.equal({
+				redsocks: {
+					local_ip: '123',
+					local_port: 'foo',
+					type: 'socks6',
+					ip: '456',
+					port: 'bar',
+					login: 'user',
+					password: 'pass',
+				},
+			});
+		});
+
+		it('parses to empty config with warnings while some key-value pairs are malformed', () => {
+			// Examples of malformed key-value pairs:
+			// - no equals sign
+			// - no value
+			// - no key
+			// - no ending semicolon (will log a warning but not prevent parsing as key and value are present)
+			const redsocksConfStr = stripIndent`
+                base {
```

Comment:
We don't need to parse the `base` block

Reply:
Amended all relevant commits to remove base configuration handling. `RedsocksConf.parse` now only parses the writeable subset of `redsocks {...}` block in redsocks.conf in 0ce259c1845c9137b07618aa6aafa54361771693, and `RedsocksConf.stringify` adds the base block statically in 9e89091ce35dbf010ada7e8167c3146acaf4ca3c.